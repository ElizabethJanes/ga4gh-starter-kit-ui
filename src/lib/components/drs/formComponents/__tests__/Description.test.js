import React from 'react';
import renderer from 'react-test-renderer';
import Description from '../Description';

test('SHOW <Description /> should handle null description', () => {
    const description = renderer
    .create(<Description readOnly={true} 
        description={null} />)
    .toJSON();
    expect(description).toMatchSnapshot();
});

test('SHOW <Description /> should handle populated description', () => {
    const description = renderer
    .create(<Description readOnly={true} 
        description={'This is a snapshot test sample description.'} />)
    .toJSON();
    expect(description).toMatchSnapshot();
});

test('NEW and EDIT <Description /> should handle null description', () => {
    const description = renderer
    .create(<Description readOnly={false} 
        description={null} />)
    .toJSON();
    expect(description).toMatchSnapshot();
});

test('NEW and EDIT <Description /> should handle populated description', () => {
    const description = renderer
    .create(<Description readOnly={false} 
        description={'This is a snapshot test sample description.'} />)
    .toJSON();
    expect(description).toMatchSnapshot();
});