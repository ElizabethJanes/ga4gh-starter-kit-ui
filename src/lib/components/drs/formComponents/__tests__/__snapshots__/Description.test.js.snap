// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <Description /> should handle null description 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="description"
      id="description-label"
    >
      Description
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="description"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
        value={null}
      />
    </div>
    <p
      className="MuiFormHelperText-root"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;

exports[`NEW and EDIT <Description /> should handle populated description 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="description"
      id="description-label"
    >
      Description
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="description"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
        value="This is a snapshot test sample description."
      />
    </div>
    <p
      className="MuiFormHelperText-root"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;

exports[`SHOW <Description /> should handle null description 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="description"
      id="description-label"
    >
      Description
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="description"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value={null}
      />
    </div>
    <p
      className="MuiFormHelperText-root"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;

exports[`SHOW <Description /> should handle populated description 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="description"
      id="description-label"
    >
      Description
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="description"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="This is a snapshot test sample description."
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-filled"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;
